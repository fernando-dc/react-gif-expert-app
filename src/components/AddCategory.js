import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState("");
    
    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    };
    const handleSubmit = (e) =>{
        e.preventDefault();
        //setCategories will use the callback function to access the array elements since this component cant do that
        if (inputValue.trim().length > 2){
            setCategories( categories => [inputValue, ...categories]);
            setinputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input 
                className='animate__animated animate__fadeInDown animate__delay-1s'
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                placeholder='Input new value'
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
    