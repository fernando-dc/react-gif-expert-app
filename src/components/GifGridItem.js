import React from 'react';

export const GifGridItem = ({id, title, url}) => {
    console.log(id, title, url);
    return(
        <div className='card animate__animated animate__fadeIn animate__slow'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};
