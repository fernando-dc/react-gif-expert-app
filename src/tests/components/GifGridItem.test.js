import React from 'react';
import '@testing-library/jest-dom';
import {shallow, } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Testing for GifGridItem', () => {
    const id = '';
    const title = 'A title';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />) 
    
    test('Display component correctly', () => {


        expect(wrapper).toMatchSnapshot();
    });
    
    test('Display component title', () => {
        const p = wrapper.find('p').text().trim();

        expect(p).toBe(title)
    });
    
    test('Display image properties', () => {
        const image = wrapper.find('img');

        expect(image.prop('src')).toBe(url);
        expect(image.prop('alt')).toBe(title);
    });
    
    test('Grid item must have "animate__fadeIn" class', () => {
        const div = wrapper.find('div');

        expect(div.hasClass('animate__fadeIn')).toBe(true);
    });
    
});
