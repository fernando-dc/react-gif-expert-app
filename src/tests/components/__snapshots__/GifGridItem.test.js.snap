// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for GifGridItem Display component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__slow"
>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
